pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -919,6 +919,46 @@ The max allowed delay for delayed delivery (in milliseconds). If the broker rece
     )
     private int brokerMaxConnections = 0;
 
+    @FieldContext(
+        category = CATEGORY_POLICIES,
+        doc = "It relates to configuration \"WriteBufferHighWaterMark\" of Netty Channel Config. If the number of bytes"
+            + " queued in the write buffer exceeds this value, channel writable state will start to return \"false\"."
+    )
+    private int pulsarChannelWriteBufferHighWaterMark = 64 * 1024;
+
+    @FieldContext(
+        category = CATEGORY_POLICIES,
+        doc = "It relates to configuration \"WriteBufferLowWaterMark\" of Netty Channel Config. If the number of bytes"
+                + " queued in the write buffer is smaller than this value, channel writable state will start to return"
+                + " \"true\"."
+    )
+    private int pulsarChannelWriteBufferLowWaterMark = 32 * 1024;
+
+    @FieldContext(
+        category = CATEGORY_POLICIES,
+        doc = "If enabled, the channel will pause to deal with new request once the writer buffer is full, until it"
+                + " is changed to writable."
+    )
+    private boolean pulsarChannelPauseReceivingRequestsIfUnwritable = false;
+
+    @FieldContext(
+            category = CATEGORY_POLICIES,
+            doc = "After the connection is recovered from an unreadable state, the channel will be rate-limited for a"
+                    + " period of time to avoid overwhelming due to the backlog of requests. This parameter defines how"
+                    + " many requests should be allowed in the rate limiting period."
+
+    )
+    private int pulsarChannelRateLimitingRateAfterResumeFromUnreadable = 1000;
+
+    @FieldContext(
+        category = CATEGORY_POLICIES,
+            doc = "After the connection is recovered from an unreadable state, the channel will be rate-limited for a"
+                    + " period of time to avoid overwhelming due to the backlog of requests. This parameter defines how"
+                    + " long the rate limiting should last, in seconds. Once the bytes that are waiting to be sent out"
+                    + " reach the \"pulsarChannelWriteBufferHighWaterMark\"， the timer will be reset."
+    )
+    private int pulsarChannelRateLimitingSecondsAfterResumeFromUnreadable = 5;
```

Comment:
There should be a way to disable this rate limiting while throttling ("pause receiving") is in use. Please document the values to disable it.

Reply:
Added